### DISCUSSION TITLE
Getting 'unknown_cast' error while binding ComboBox SelectedValue attribute.

### DISCUSSION THREAD

**Discussion by geniuszombie** [microsoft/microsoft-ui-xaml#7484] [Q&A]
Hi,

I am trying to bind ComboBox ItemsSource attribute as follows in my C++/WinRT project but getting the error that I have mentioned in the title.

```
<ComboBox x:Uid="ComboSample"
          ItemsSource="{x:Bind ViewModel.ComboItems, Mode=OneTime}"
          SelectedValuePath="Key"
          SelectedValue="{x:Bind ViewModel.ItemKey, Mode=TwoWay}">
    <ComboBox.ItemTemplate>
        <DataTemplate x:DataType="local:DataSourceItem" >
            <TextBlock Text="{x:Bind Text}"></TextBlock>
        </DataTemplate>
    </ComboBox.ItemTemplate>
</ComboBox>
```

ComboItems collection contains a list of object that has key and text variables with the types of  `int16_t` and `hstring`. ViewModel ItemKey variable is also an `int16_t`. With this setup I am getting following errors in generated `xaml.g.hpp` file.

```
Error	C2065	'unknown_cast': undeclared identifier
Error	C2275	'int16_t': illegal use of this type as an expression
```

Generated file content for the related code is follows,
```
if (GetDataRoot().ViewModel() != nullptr)
{
    GetDataRoot().ViewModel().ItemKey(unknown_cast<int16_t>(obj6.SelectedValue()));
}
```

Is this a bug or am I doing somethinig wrong?

Thanks.

**Comment by jaigak** [comment]
Selected value for ComboBox should be an int32_t. C++ will allow you to assign a int16_t to int32_t directly but you'll receive a warning. I don't think the XAML compiler knows this so change it to int32_t

**Comment by beckni** [comment]
Hi,
I have exactly the same issue. Changing the type to int32_t didn't resolve the problem. Now the generated code is simply

```
if (GetDataRoot() != nullptr)
{
    GetDataRoot().SelectedMaterialId(unknown_cast<int32_t>(obj6.SelectedValue()));
}
```
Also, where can I find the documentation on what data type must be used for a given attribute?

**Reply by jaigak** [reply]
I've already reported this bug of the generated code using unknown_cast, which doesn't exist: https://github.com/microsoft/microsoft-ui-xaml/issues/7986.
